uui-core/hooks/useTableState/useTableState.ts

```diff
@@ -42,66 +44,75 @@ export const useTableState = <TFilter = Record<string, any>>(params: IParams<TFi
         //         ? newValue.page
         //         : 1,
         // }));
-        
-        if (!isFilterEqual || query.presetId !== +newValue.presetId) {
-            const newQuery = {
-                ...context.uuiRouter.getCurrentLink().query,
-                filter: encodeURIComponent(JSON.stringify(newFilter)),
-                presetId: newValue.presetId,
-            };
-            if (!newValue.presetId) {
-                delete newQuery.presetId;
-            }
-
-            context.uuiRouter.redirect({
-                pathname: location.pathname,
-                query: newQuery,
-            });
+
+        const newQuery = {
+            ...context.uuiRouter.getCurrentLink().query,
+            filter: JSON.stringify(newValue.filter),
```

Comment:
do we need to have JSON.stringify here? It should be done by history adapter for full query object.